### PR TITLE
PR 1

### PR DESCRIPTION
**Pull request by dawkrish**
Put all the code of handlefunctions into /internal/handlers

### CODE REVIEW

**Comment by wipdev-tech** [comment]
I am not sure if I like the handlers being in an internal package for a couple of reasons:

- In this form, some of the "handling" happens in `package main` and some of it happens in `package handlers`, as you can see form `os.Args` being parsed in more than one package now.
- Also, instead of passing down `configDir`, `configPath` to other packages, it may have been better to keep everything localized to one package. But this is not a critical issue.

So, if you think the handler functions should be separated, then they could be better off being in a `handlers.go` file in the root directory so that it belongs to `package main` and shares the same data and purposes of parsing and handling the command-line arguments.

What do you think?

**Comment by dawkrish** [comment]
I think the idea of `handlers.go` makes more sense
but I don't understand why have packages for `config` and `fzf` after all ?

Ughh

I want to somehow have `main.go` a function `handlers.StartRepl()` and this can be achieved by wraping of all the current `main.go` into `handlers.StartRepl()` ????

**Comment by wipdev-tech** [comment]
Think of it this way: an "internal" package has "internal" functionality whose implementation details should not be exposed to `main`. So, `package config` has functions and types to manipulate the config file, and `package fzf` has all the `bubbletea` stuff to render a TUI. So, `main()` shouldn't have to concern itself about how those functions internally work - they only call them and receive values/errors and act accordingly.

For example, I could change the whole config structure to be XML or some other format instead of JSON, and I wouldn't have to change a thing in `main` or `fzf` (or so I hope) because the related logic is contained only in `config`.

Now if you wrapped all the logic in `main` inside an internal package so that you only call a single function within `main`, then what did you benefit?

In other words, if your program works like:
1. `main` only calls `handlers.StartRepl` and nothing else
2. `handlers.StartRepl` does the handling logic

Then why not leave the handling logic at `main` it's the only thing that happens there anyway?
Does that make sense?

**Comment by dawkrish** [comment]
Absolutley makes sense
the example you give was very good
now I am closing the PR
Thanks alot big Waseem
